### DISCUSSION TITLE
Add "modern" WASD keybinds to an opcional CFG

### DISCUSSION THREAD

**Discussion by sbritorodr** [Novum/vkQuake#445] [Ideas]
First of all, thank you for creating this awesome quake1 engine.

Each time I install this engine I need to re-bind WASD for just playing the game. I think it will be nice if there's a preconfig file included in the instalation files. Like a modern.cfg o thresh.cfg that could be accessed optionally by `exec thresh.cfg` or `exec modern.cfg`.

This could be the cfg file:
```cfg

//Movement keys 
bind w "+forward" 
bind a "+moveleft" 
bind d "+moveright" 
bind s "+back" 
bind SHIFT "+speed" 
bind SPACE "+jump" 
bind ALT "+movedown" 

// Mouse Settings
+mlook
lookstrafe 0
bind MOUSE1 "+attack"

//Weapons
bind "0" "impulse 0"
bind "1" "impulse 1"
bind "2" "impulse 2"
bind "3" "impulse 3"
bind "4" "impulse 4"
bind "5" "impulse 5"
bind "6" "impulse 6"
bind "7" "impulse 7"
bind "8" "impulse 8"

//Function keys
bind "F6" "echo Quicksaving...; wait; save quick"
bind "F9" "echo Quickloading...; wait; load quick"

//Misc
bind "TAB" "+showscores"
bind "ESCAPE" "togglemenu"


```

**Comment by Novum** [comment]
This is the `default.cfg` that ships with vkQuake (copied from QuakeSpasm)

```
//
// load keybindings
//
// commands with a leading + will also be called for key up events with
// the + changed to a -
unbindall

//
// character controls
//

bind	ALT				+strafe

bind	,				+moveleft
bind	a				+moveleft
bind	.				+moveright
bind	d				+moveright
bind	DEL				+lookdown
bind	PGDN			+lookup
bind	END				centerview

bind	e				+moveup
bind	c				+movedown
bind	SHIFT			+speed
bind	CTRL			+attack
bind	UPARROW			+forward
bind	w			+forward
bind	DOWNARROW		+back
bind	s			+back
bind	LEFTARROW		+left
bind	RIGHTARROW		+right

bind	SPACE			+jump 
//bind	ENTER			+jump

bind	TAB				+showscores

bind	1				"impulse 1"
bind	2				"impulse 2"
bind	3				"impulse 3"
bind	4				"impulse 4"
bind	5				"impulse 5"
bind	6				"impulse 6"
bind	7				"impulse 7"
bind	8				"impulse 8"

bind	0				"impulse 0"

bind	/				"impulse 10"		// change weapon
bind	MWHEELDOWN			"impulse 10"
bind	MWHEELUP			"impulse 12"

// zoom
alias zoom_in "sensitivity 2;fov 90;wait;fov 70;wait;fov 50;wait;fov 30;wait;fov 10;wait;fov 5;bind F11 zoom_out"
alias zoom_out "sensitivity 4;fov 5;wait;fov 10;wait;fov 30;wait;fov 50;wait;fov 70;wait;fov 90;bind F11 zoom_in; sensitivity 3"
bind F11 zoom_in

// Function keys
bind	F1			"help"
bind	F2			"menu_save"
bind	F3			"menu_load"
bind	F4			"menu_options"
bind	F5			"menu_multiplayer"
bind	F6			"echo Quicksaving...; wait; save quick"
bind	F9			"echo Quickloading...; wait; load quick"
bind	F10			"quit"
bind    F12                     "screenshot"

// mouse options
bind	\				+mlook

//
// client environment commands
//
bind	PAUSE			"pause"
bind	ESCAPE		"togglemenu"
bind	~				"toggleconsole"
bind	`				"toggleconsole"

bind	t				"messagemode"

bind	+				"sizeup"
bind	=				"sizeup"
bind	-				"sizedown"

bind	INS			+klook

//
// mouse buttons
//
bind	MOUSE1		+attack
//bind	MOUSE2		+forward
bind	MOUSE2		+jump
//bind	MOUSE3		+mlook

//
// game controller
//
bind	LSHOULDER	"impulse 12"
bind	RSHOULDER	"impulse 10"
bind	LTRIGGER	+jump
bind	RTRIGGER	+attack

//
// default cvars
//
gamma 			1.0
volume 			0.7
sensitivity 	3

//viewsize 		100
viewsize		110
scr_conscale	1.6
scr_menuscale	1.6
scr_sbarscale	1.6

// default to mouse-look enabled
+mlook
```

If things are different to this it's because of already existing cfg files.

**Comment by ghost** [comment]
For me it already had WASD controls when I installed it. Either you replaced the CFG, or you just saw that comma, period, and the arrow keys were bound and immediately put up a discussion and didn't actually bother or check to see that WASD was also bound.

**Comment by sbritorodr** [comment]
This was created almost 40 days ago and right know I can't test the 22 dec build in my windows machine. For some reason at that time the game didn't bind the default controls and I needed to re-bind it myself. I've just downloaded and compiled in a MacOS machine a version prev to March 24 and works as you said. Sorry for the inconvenience.

**Reply by ghost** [reply]
> This was created almost 40 days ago and right know I can't test the 22 dec build in my windows machine. For some reason at that time the game didn't bind the default controls and I needed to re-bind it myself. I've just downloaded and compiled in a MacOS machine a version prev to March 24 and works as you said. Sorry for the inconvenience.

I just downloaded the December build and it still has WASD.

**Comment by Novum** [comment]
The problem is that e.g. the GOG release of Quake is a complete mess and has files there (including .cfg files) that should not be there and overwrite the default config of vkQuake.

There isn't much I can do about that.

**Reply by ghost** [reply]
> The problem is that e.g. the GOG release of Quake is a complete mess and has files there (including .cfg files) that should not be there and overwrite the default config of vkQuake.
> 
> There isn't much I can do about that.

That makes sense. Couldn't you have the config file instead named vkconfig.cfg or something along those lines and point the game to that instead? I am no programmer so there is probably a million things wrong with what I just said so apologies.

**Comment by Novum** [comment]
Not really, the files that are included are supposed to override the defaults by design.
